src/NuGet.Core/NuGet.Common/Telemetry/TelemetryEvent.cs

```diff
@@ -8,36 +8,36 @@ namespace NuGet.Common
     /// <summary> Represents a NuGet telemetry event data to pass to telemetry service. </summary>
     public class TelemetryEvent
     {
-        private IDictionary<string, object> _properties;
-        private IDictionary<string, object> _piiProperties;
+        private IDictionary<string, object?> _properties;
+        private IDictionary<string, object?> _piiProperties;
 
         /// <summary> Creates a new instance of <see cref="TelemetryEvent"/>. </summary>
         /// <param name="eventName"> Event name. </param>
-        public TelemetryEvent(string eventName) :
-            this(eventName, new Dictionary<string, object>())
+        public TelemetryEvent(string? eventName) :
+            this(eventName, new Dictionary<string, object?>())
         {
         }
 
         /// <summary> Creates a new instance of <see cref="TelemetryEvent"/>. </summary>
         /// <param name="eventName"> Event name. </param>
         /// <param name="properties"> Properties to add to the event. </param>
-        public TelemetryEvent(string eventName, Dictionary<string, object> properties)
+        public TelemetryEvent(string? eventName, Dictionary<string, object?> properties)
```

Comment:
Why is the value nullable?

Reply:
Say we have telemetry saying `telemetryEvent["SomeMSBuildProperty"] = value` to tell us what value customers are using for some feature. If the project doesn't define a value for that MSBuild property, it makes sense for the property to be null, and it also makes sense for us to want telemetry telling us about the null value, rather than omitting the telemetry property completely.